Mage/src/main/java/mage/abilities/effects/common/ClashEffect.java

```diff
@@ -181,7 +181,18 @@ public boolean apply(Game game, Ability source) {
                     if (cardOpponent != null) {
                         opponent.moveCardToLibraryWithInfo(cardOpponent, source.getSourceId(), game, Zone.LIBRARY, topOpponent, true);
                     }
-                    game.fireEvent(new GameEvent(EventType.CLASHED, opponent.getId(), source.getSourceId(), controller.getId(), 0, cmcController > cmcOpponent));
+                    // fire CLASHED event with info about who won
+                    String winner = "draw";
+                    if (cmcController > cmcOpponent) {
+                        winner = "controller";
+                    }
+                    if (cmcOpponent > cmcController) {
+                        winner = "opponent";
+                    }
+                    GameEvent gameEvent = new GameEvent(EventType.CLASHED, opponent.getId(), source.getSourceId(), controller.getId());
+                    gameEvent.setData(winner);
+                    game.fireEvent(gameEvent);
+

```

Comment:
No, it doesn't. I just changed the event to determine the winner (before you could only determine if the source controller won). Only card yet implemented that used the event was EntanglingTrap which I fixed.